__tests__/e2e/synthetics.journey.ts

```diff
@@ -0,0 +1,208 @@
+  await page.click('[data-test-subj="loginSubmit"]');
+}
+
+async function goToSyntheticsIntegrationPage(page) {
+  await page.goto(`http://localhost:5601/app/fleet/integrations/synthetics-${syntheticsPackageVersion}/add-integration`);
+  await page.waitForSelector('[data-test-subj="loginUsername"]', { timeout: 10000 });
+  const isUnauthenticated = await page.isVisible('[data-test-subj="loginUsername"]');
+  if (isUnauthenticated) {
+    await logIn(page);
+  }
+  await page.waitForTimeout(10 * 1000);
+}
+
+async function goToUptime(page) {
+  await page.goto('http://localhost:5601/app/uptime');
+  await page.waitForTimeout(10 * 1000);
+}
+
+async function createIntegrationPolicyName({ page, policyName }) {
+  await page.waitForSelector('[data-test-subj="packagePolicyNameInput"]', { timeout: 10000 });
+  await page.fill('[data-test-subj="packagePolicyNameInput"]', policyName);
+}
+
+async function confirmAndSavePolicy(page) {
+  await page.click('[data-test-subj="createPackagePolicySaveButton"]');
+  await page.click('[data-test-subj="confirmModalConfirmButton"]');
+  await page.waitForSelector('[data-test-subj="packagePolicyCreateSuccessToast"]', { timeout: 20000 });
+}
+
+async function checkForSyntheticsData({ page, journeyName }) {
+  await page.click(`text=${journeyName}`, { timeout: 240 * 1000 });
+  await page.waitForTimeout(10 * 1000);
+  
+  const content = await page.textContent('h1');
+  expect(content).toBe(journeyName);
+}
+
+journey('E2e test synthetics - http', async ({ page }) => {
+  const journeyName = 'Sample http integration policy';
+
+  step('Go to synthetics integration page', async () => {
+    await goToSyntheticsIntegrationPage(page);
+  });
+
+  step('create an http monitor', async () => {
+    await createIntegrationPolicyName({ page, policyName: journeyName });
+    await page.fill('[data-test-subj="syntheticsUrlField"]', 'https://elastic.co');
+    await confirmAndSavePolicy(page);
+  });
+
+  step('go to uptime', async () => {
+    await goToUptime(page);
+  });
+
+  step('wait for synthetics data', async () => {
+    await checkForSyntheticsData({ page, journeyName });
+  });
+});
+
+journey('E2e test synthetics - tcp', async ({ page }) => {
+  const journeyName = 'Sample tcp integration policy';
+
+  step('Go to synthetics integration page', async () => {
+    await goToSyntheticsIntegrationPage(page);
+  });
+
+  step('create an tcp monitor', async () => {
+    await createIntegrationPolicyName({ page, policyName: journeyName });
+    await page.selectOption('[data-test-subj="syntheticsMonitorTypeField"]', 'tcp');
+    await page.fill('[data-test-subj="syntheticsTCPHostField"]', 'smtp.gmail.com:587');
+    await confirmAndSavePolicy(page);
+  });
+
+  step('go to uptime', async () => {
+    await goToUptime(page);
+  });
+
+  step('wait for synthetics data', async () => {
+    await checkForSyntheticsData({ page, journeyName });
+  });
+});
+
+// journey('E2e test synthetics - icmp', async ({ page }) => {
```

Comment:
ICMP tests are currently disabled, as icmp requires `elastic-agent` to be run as root.